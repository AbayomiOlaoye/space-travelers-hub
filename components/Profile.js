import React from 'react';
/* import { Link } from 'react-router-dom'; */

const Profile = () => {
  <header>
    {/* <div className="profile">
      <Link>Make Reservations</Link>
    </div> */}
  </header>;
};

export default Profile;
