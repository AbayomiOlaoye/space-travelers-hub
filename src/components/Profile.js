import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import MissionProfiler from './MissionProfiler';

const Profile = () => {
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocks) => rocks.active);

  return (
    <div className="profile">
      <div className="rockets">
        <h1 className="head-text">My Rockets</h1>
        {reservedRockets.length === 0 ? (
          <div className="noReservations">
            <p>No Reservations Made</p>
            <Link className="reserveRocketsLink" to="/">Make Reservations</Link>
          </div>
        ) : (
          <div className="activeRockets">
            <ul>
              {reservedRockets.map((rocket) => (
                <li key={rocket.id}>{rocket.rocket_name}</li>
              ))}
            </ul>
          </div>
        )}
      </div>
      <MissionProfiler />
    </div>
  );
};

/* import MyProfile from './MissionProfiler';

const Profile = () => (
  <div className="container">
    <MyProfile />
  </div>
); */

export default Profile;
