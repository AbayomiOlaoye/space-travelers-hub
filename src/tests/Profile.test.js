import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Profile from '../components/Profile';
import store from '../redux/configureStore';

jest.setTimeout(10000);
describe('Profile NavLink', () => {
  it('Should render a list', () => {
    const profileLink = render(
      <>
        <Provider store={store}>
          <Profile />
        </Provider>
      </>,
    );
    expect(profileLink).toMatchSnapshot();
  });
});
