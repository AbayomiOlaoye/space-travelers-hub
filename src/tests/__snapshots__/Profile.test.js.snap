// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile section must render a list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div className="container">
      <div>
        <h1 className="head-text">My Missions</h1>
        <table className="tour_list">
          <tbody className="tour_body" />
        </table>
      </div>
      <div className="profile">
        <div className="rockets">
          <h1>My Rockets</h1>
          <table className="tour-now tour">
            <td>No Reservations Made...</td>
            <td><Link className="reserveRocketsLink" to="/">Make a Reservation</Link></td>
          </table>
          <div className="activeRockets">
            <ul>
              <li key=""></li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    </body>,
  "container": <div>
    <div>
      <h1 className="head-text">My Missions</h1>
      <table className="tour_list">
        <tbody className="tour_body" />
      </table>
    </div>
    <div className="profile">
      <div className="rockets">
        <h1>My Rockets</h1>
        <table className="tour-now tour">
          <td>No Reservations Made...</td>
          <td><Link className="reserveRocketsLink" to="/">Make a Reservation</Link></td>
        </table>
        <div className="activeRockets">
          <ul>
            <li key=""></li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
